app/internal/astronomy.py

```diff
@@ -0,0 +1,88 @@
+import datetime
+import functools
+import frozendict
+import requests
+from typing import Dict, Tuple, Union
+
+from app import config
+
+
+# This feature requires an API KEY - get yours free @ www.weatherapi.com
+
+SUCCESS_STATUS = 0
+ERROR_STATUS = -1
+ASTRONOMY_URL = "http://api.weatherapi.com/v1/astronomy.json"
+NO_API_RESPONSE = "No response from server"
+
+
+def freezeargs(func):
+    """Transform mutable dictionary into immutable
+    Credit to 'fast_cen' from 'stackoverflow'
+    https://stackoverflow.com/questions/6358481/
+    using-functools-lru-cache-with-dictionary-arguments
+    """
+    @functools.wraps(func)
+    def wrapped(*args, **kwargs):
+        args = tuple([frozendict.frozendict(arg)
+                      if isinstance(arg, dict) else arg for arg in args])
+        kwargs = {k: frozendict.frozendict(v) if isinstance(v, dict) else v
+                  for k, v in kwargs.items()}
+        return func(*args, **kwargs)
+    return wrapped
+
+
+@freezeargs
```

Comment:
Why we need the freezeargs?

Reply:
After following your advices it seems that it is no longer necessary.
I've removed it :-)